### PR TITLE
When shutting down the runtime we should always close the database

### PR DESCRIPTION
**Pull request by rhatdan**
Even if the storage fails to shutdown. This patch fixes on TODO.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mheon PTAL

File: libpod/runtime.go [hunk 1]
```diff
@@ -227,9 +227,8 @@ func (r *Runtime) Shutdown(force bool) error {
 
 	_, err := r.store.Shutdown(force)
 	if err != nil {
-		return err
+		logrus.Errorf("Error shutting down storage: %v", err)
```
Comment (mheon) [inline libpod/runtime.go]: Maybe save this as lastError similar to the code in cmd/kpod, so we don't exit cleanly if r.state.Close() succeeds but r.store.Shutdown() fails

**Comment by TomSweeneyRedHat** [comment]
Travis seems unhappy with go 1.8, but passes with go 1.9?
```
not ok 28 kill a running container by id with a bogus signal
# (in test file ./kpod_kill.bats, line 61)
#   `[ "$status" -eq 1 ]' failed
# 203c64e8607ff754c8da61468131cd17bb0af259be50d13bac829a6d8e6adf90
```

**Comment by rhatdan** [comment]
Tests are passing.
PTAL @mheon @TomSweeneyRedHat @baude @umohnani8 And merge.

File: libpod/runtime.go [hunk 2]
```diff
@@ -225,11 +225,16 @@ func (r *Runtime) Shutdown(force bool) error {
...
-		return err
+	var lastError error
+	if _, err := r.store.Shutdown(force); err != nil {
+		lastError = errors.Wrapf(err, "Error shutting down container storage")
+	}
+	if err := r.state.Close(); err != nil {
+		if lastError != nil {
+			logrus.Errorf("%v", lastError)
+		}
+		lastError = err
```
Comment (TomSweeneyRedHat) [inline libpod/runtime.go]: This maybe golang education time for me, but a lot of these calls to this function are:
```
defer runtime.Shutdown(false)
```
Will this second lastError get printed in that instance or do we need a second logrus.Errorf statement here?

**Comment by umohnani8** [comment]
LGTM

File: libpod/runtime.go (hunk 2 above)
Comment (mheon) [reply libpod/runtime.go]: I think this is more an error in the defer statement than it is one in the library - if the error is being thrown out there's not much we can do.
I thought Golint would catch ignored returns in deferred functions, but apparently not.
Should see if there's a way to enable that.

**Comment by mheon** [comment]
LGTM
@rh-atomic-bot r+

**Pull request closed without merging**